### ISSUE TITLE
XAxis Object missing Attributes

### ISSUE THREAD

**Issue by spprichard** [weaveworks/grafanalib#124]
## What you expected to happen?
When creating an `XAxis` Object expect more then just the `show` attribute on the XAxis Object to be translated into JSON. 
## What happened?

There is no error message in this case. However, if you look at the JSON that is generated, only the `show` key is set. 
```json
"xaxis": {
    "show": true
}, 
```

I expect the following:
```json
"xaxis": {
    "mode": "series", 
     "name": null, 
     "show": true, 
     "values": [
         "current"
     ]
}, 
```

## How to reproduce it?

- Create a Graph Object
- Give it a `xAxis` attribute 
- Create a `XAxis` _see below_
```json
xAxis=XAxis(
    show=True,
    mode="series",
    values=["current"],
),
```
- Observe generated JSON
```json
"xaxis": {
     "show": true
}, 
```

I would propose that the XAxis Class should look like this:
```python
@attr.s
class XAxis(object):

    mode = attr.ib(default="time", validator=is_valid_xaxis_mode)
    name = attr.ib(default=None)
    values = attr.ib(default=attr.Factory(list))
    show = attr.ib(validator=instance_of(bool), default=True)

    def to_json_data(self):
        return {
            'show': self.show,
            'mode': self.mode,
            'values': self.values,
            'name' : self.name,
        }
```

**Comment by ushasm** [comment]
Even I am face the same issue. 
Can we commit the changes and create PR for the same ?

**Comment by jml** [comment]
PRs are very welcome!

**Comment by ushasm** [comment]
Seems like we do not have permission to create a feature branch and commit the changes :(

**Comment by 3v1lm1ky** [comment]
I have same issue right here

**Comment by JamesGibo** [comment]
Closed in #369
